setup.py

```diff
@@ -87,7 +84,11 @@ def package_files(directory):
 setup(
     name='cocotb',
     cmdclass={'build_ext': build_ext},
-    version=__version__,  # noqa: F821
+    use_scm_version={
+        'write_to': 'cocotb/_version.py',
+        'write_to_template': '__version__ = "{version}"',
```

Comment:
nit: 
```suggestion
        'write_to_template': '__version__ = {version!r}',
```
although it only makes a difference if the version contains quotes or special characters.